### PR TITLE
Autonomous

### PR DESCRIPTION
**Pull request by impossiblerocket**
added autonomous (commented stuff in robot.java is for aiden to merge in his stuff later easily)

### CODE REVIEW

File: TitanRobotics2024/src/main/java/frc/robot/Auto/AutoMissionChooser.java [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+import edu.wpi.first.wpilibj.smartdashboard.SendableChooser;
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
+
+public class AutoMissionChooser {
+      enum DesiredMode {
+        doNothing,
+        leaveCommunityRight,
+        exampleMission
+    }
+
```
Comment (thebuilder6) [inline TitanRobotics2024/src/main/java/frc/robot/Auto/AutoMissionChooser.java]: Standardize the enum values to uppercase
EXAMPLE_MISSION

File: TitanRobotics2024/src/main/java/frc/robot/Auto/AutoMissionChooser.java [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+     private Optional<MissionBase> getAutoModeForParams(DesiredMode mode) {
+        switch (mode) {
+            case doNothing:
+                return Optional.of(new DoNothingMission());
+            case leaveCommunityRight:
+                return Optional.of(new LeaveCommunityRightMission());
+            case exampleMission:
+                return Optional.of(new ExampleMission());
+            default:
+                break;
```
Comment (thebuilder6) [inline TitanRobotics2024/src/main/java/frc/robot/Auto/AutoMissionChooser.java]: Remove the unnecessary break statement, and make an informative error message
```
default:
   System.err.println("No valid auto mode found for " + mode);
   return Optional.empty();
```

File: TitanRobotics2024/src/main/java/frc/robot/Robot.java
Comment (thebuilder6) [inline TitanRobotics2024/src/main/java/frc/robot/Robot.java:1]: Consider removing all coments and unused code to reduce size to more managable levels

File: TitanRobotics2024/src/main/java/frc/robot/Teleop/Teleop.java
Comment (thebuilder6) [inline TitanRobotics2024/src/main/java/frc/robot/Teleop/Teleop.java:1]: Still not sure what this file/class is good for

**Review by thebuilder6 — CHANGES_REQUESTED** [review]

File: TitanRobotics2024/src/main/java/frc/robot/Auto/AutoMissionChooser.java (hunk 1 above)
Comment (impossiblerocket) [reply TitanRobotics2024/src/main/java/frc/robot/Auto/AutoMissionChooser.java]: better/ easier to read in the current state

**Review by thebuilder6 — APPROVED** [review]

File: TitanRobotics2024/src/main/java/frc/robot/Auto/AutoMissionChooser.java (hunk 1 above)
Comment (thebuilder6) [reply TitanRobotics2024/src/main/java/frc/robot/Auto/AutoMissionChooser.java]: I think the Constants and Enum values should have a way to differentiate them from other variables, but I have approved your changes

**Pull request merged**